cli/node/mod.rs

```diff
@@ -425,12 +479,22 @@ pub fn node_resolve(
     }
   }
 
-  let conditions = DEFAULT_CONDITIONS;
+  let conditions = mode_conditions(mode);
   let url = module_resolve(specifier, referrer, conditions, npm_resolver)?;
   let url = match url {
     Some(url) => url,
     None => return Ok(None),
   };
+  let url = match mode {
+    NodeResolutionMode::Execution => url,
+    NodeResolutionMode::Types => {
+      let path = url.to_file_path().unwrap();
```

Comment:
Are you 100% sure that it will always be a file path? If so, please add a comment about it

Reply:
I think we routininely do this in the node code though?